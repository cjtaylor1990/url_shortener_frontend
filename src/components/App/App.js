import React from 'react';
import './App.css';

import LoginForm from '../LoginForm/LoginForm.js';
import UrlForm from '../UrlForm/UrlForm.js';

class App extends React.Component {
	constructor(props) {
		super(props)
		
		this.state = {
			authenticated: false,
			username: undefined,
			password: undefined
		};

		this.authenticate = this.authenticate.bind(this);
	}

	authenticate(username,password){
		if (username && password) {
			this.setState({
				authenticated: true,
				username: username,
				password: password
			});
		};
	}

	render() {
    	if (this.state.authenticated) {
			return (
				<div className="App">
					<UrlForm />
				</div>
      	);
    	} else {
      		return (
        		<div className="App">
					<LoginForm authenticate={this.authenticate} />
				</div>
      		);
    	};
	};

}

export default App
