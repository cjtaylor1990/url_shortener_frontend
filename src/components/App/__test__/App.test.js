import React from 'react'
import ReactDOM from 'react-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import App from '../App.js';

afterEach(cleanup);

describe('User initially accesses App', () => {
    
    it('renders App without crashing', () => {
        const div = document.createElement('div');
        
        ReactDOM.render(<App />, div);
        
        ReactDOM.unmountComponentAtNode(div);
    });

    it('App has displays a LoginForm', () => {
        const { getByTestId } = render(<App/>);
        
        const renderedLoginForm = getByTestId('loginForm');
        
        expect(renderedLoginForm).toBeInTheDocument();
    });

});

describe('User submits username and password', () => {

    it('Username and password are recognized. App renders UrlForm', () => {
        const { getByTestId } = render(<App />);
    
        const usernameInput = getByTestId('loginFormUsername');
        const passwordInput = getByTestId('loginFormPassword');
        const loginButton = getByTestId('loginFormLogin');

        const username = 'user1234';
        const password = 'password1';
        fireEvent.change(usernameInput, {target: {value: username}});
        fireEvent.change(passwordInput, {target: {value: password}});

        const leftClick = {button: 1};
        fireEvent.click(loginButton, leftClick);

        const renderedUrlForm = getByTestId('urlForm');
        expect(renderedUrlForm).toBeInTheDocument();

    });

});


