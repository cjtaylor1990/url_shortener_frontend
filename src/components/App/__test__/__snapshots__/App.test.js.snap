// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User initially accesses App matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="loginForm"
    data-testid="loginForm"
  >
    <input
      className="username"
      data-testid="loginFormUsername"
      onChange={[Function]}
      placeholder="Username"
      type="text"
    />
    <input
      className="password"
      data-testid="loginFormPassword"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <button
      className="login"
      data-testid="loginFormLogin"
      onClick={[Function]}
      type="submit"
    >
       Log In 
    </button>
  </div>
</div>
`;
